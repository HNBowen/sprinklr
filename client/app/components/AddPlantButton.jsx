import React from 'react'

const AddPlantButton  = (props) => {
  return (
      <button onClick={props.displayModal}>Add A Plant</button>
    )
}

export default AddPlantButton