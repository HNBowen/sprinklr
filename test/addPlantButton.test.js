import React from 'react';
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'

import AddPlantButton from '../client/app/components/AddPlantButton.jsx'

describe('AddPlantButton', function() {

  it('should render correctly', function() {
    const wrapper = shallow(<AddPlantButton />)
    jestExpect(shallowToJson(wrapper)).toMatchSnapshot();
  })

  it('should call displayModal when clicked', function() {
    const mockDisplayModal = jest.fn()
    const wrapper = shallow(<AddPlantButton displayModal={mockDisplayModal}/>);

    wrapper.find("button").simulate("click")

    expect(mockDisplayModal.mock.calls.length).to.equal(1);

  })
})