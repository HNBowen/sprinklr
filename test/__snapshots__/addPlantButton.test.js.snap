// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPlantButton should render correctly 1`] = `
<button>
  Add A Plant
</button>
`;
